src/parse/plan.rs

```diff
@@ -0,0 +1,183 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//! Parsing of [proto::Plan].
+
+use super::{
+    extensions::{
+        simple_extension_declaration::{
+            SimpleExtensionDeclaration, SimpleExtensionDeclarationError,
+        },
+        simple_extension_uri::{SimpleExtensionURI, SimpleExtensionURIError},
+    },
+    version::{Version, VersionError},
+    Context, Parse,
+};
+use crate::proto;
+use thiserror::Error;
+
+/// A parsed [proto::Plan].
```

Comment:
From a naming standpoint I'd consider the plan to be parsed once you use the protobuf parse methods to deserialize it.  As a second step (perhaps called study) it makes more sense to me to have the validation that is present in this PR.  I could see even more validation as an optional third step, like checking for issues with output mappings, function name checking, etc.